### PR TITLE
Fix date serialization

### PR DESCRIPTION
**Pull request by gbrian**
JSON: Prevents "date" objects to fail with serialization

### CODE REVIEW

File: tests/utils_tests.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from datetime import datetime, date
+from caravel import utils
+import unittest
+
+
+class UtilsTestCase(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(UtilsTestCase, self).__init__(*args, **kwargs)
```
Comment (xrmx) [inline tests/utils_tests.py]: Do you really need this?

File: tests/utils_tests.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from datetime import datetime, date
+from caravel import utils
+import unittest
+
+
+class UtilsTestCase(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(UtilsTestCase, self).__init__(*args, **kwargs)
+
+    def test_jsonDateSerialization(self):
```
Comment (xrmx) [inline tests/utils_tests.py]: what about test_json_int_dttm_ser instead?

**Comment by gbrian** [comment]
@xrmx Sorry not that good in Python. Please feel free to change accordingly to the project standards. I'll take note for the next one.

**Comment by xrmx** [comment]
@gbrian i can't change it :)

**Comment by gbrian** [comment]
Ops, done! :)

**Comment by xrmx** [comment]
@gbrian  Thanks a lot!

File: caravel/utils.py [hunk 3]
```diff
@@ -310,6 +310,8 @@ def json_int_dttm_ser(obj):
     val = base_json_conv(obj)
     if val is not None:
         return val
+    if isinstance(obj, date):
+        obj = datetime.combine(obj, datetime.min.time())
     if isinstance(obj, datetime):
```
Comment (mistercrunch) [inline caravel/utils.py]: elif

Comment (gbrian) [inline caravel/utils.py]: Fix transforms data before serializing (date -> datetime)
Do you prefer **elif** way ?

Comment (mistercrunch) [inline caravel/utils.py]: well it's supposed to hit that `else:` if the type is not recognized which it won't now. Also no need to evaluate whether it's a `datetime` if we now its a `date`

**Comment by gbrian** [comment]
@mistercrunch: ok, gotcha and updated.

**Comment by gbrian** [comment]
@xrmx done and thanks

File: tests/utils_tests.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+from caravel import utils
+import unittest
+
+
+class UtilsTestCase(unittest.TestCase):
+    def test_json_int_dttm_ser(self):
+        today = date.today()
+        now = datetime.now()
+        ms = utils.json_int_dttm_ser(today)
+        deser = (utils.EPOCH + timedelta(milliseconds=ms)).date()
```
Comment (xrmx) [inline tests/utils_tests.py]: What does deser means? :)

Comment (xrmx) [inline tests/utils_tests.py]: yep, expected sounds better :)

**Comment by mistercrunch** [comment]
![img] (http://reaction.club/r/a0b8a78.gif)

**Pull request merged**